src/libraries/System.ComponentModel.Primitives/src/System.ComponentModel.Primitives.csproj

```
@@ -35,6 +35,9 @@
     <Compile Include="System\ComponentModel\RefreshProperties.cs" />
     <Compile Include="System\ComponentModel\RefreshPropertiesAttribute.cs" />
   </ItemGroup>
+  <ItemGroup>
+    <ILLinkLinkAttributesXmls Include="$(MSBuildThisFileDirectory)ILLink\ILLink.LinkAttributes.xml" Condition="'$(Platform)' == 'wasm' or '$(TargetsiOS)' == 'true' or '$(TargetstvOS)' == 'true' or '$(TargetsAndroid)' == 'true'" />
```

```suggestion
    <ILLinkLinkAttributesXmls Include="$(MSBuildThisFileDirectory)ILLink\ILLink.LinkAttributes.xml" Condition="''$(TargetsMobile)' == 'true'" />
```

TargetsiOS, TargetsAndroid, etc, will not be set in this project as the project doesn't cross compile for those RIDs, so we can just depend on the `TargetOS` properties. 

`TargetsMobile` is `true` on any of those or Browser:

https://github.com/dotnet/runtime/blob/188a1ee22344f181224d38df91fa8d214b76a020/eng/Configurations.props#L70

---

> TargetsiOS, TargetsAndroid, etc, will not be set in this project as the project doesn't cross compile for those RIDs, so we can just depend on the TargetOS properties.

If we don't cross-compile for those RIDs, then how will this .xml file ever make its way into the shipping product only for those RIDs?

---

Because we have a vertical leg where we produce the runtime pack for those RIDs and that is based off of `TargetOS`.

---

Is that the normal way we compile for specific OS platforms? I thought we normally used `<TargetFrameworks>$(NetCoreAppCurrent);$(NetCoreAppCurrent)-someOS</TargetFrameworks>`

---

Yeah I don't know if we should cross compile for this ILlink file. I thought it would be enough with using TargetOS and having the verticals for those RIDs get the trim file on the runtime pack. But now you've made me doubt what's best. I think we should cross compile as we do elsewhere for different outputs. cc: @ericstj

---

We should only be reading properties that are controlled by the config system.  Otherwise you get different behavior when building from the root vs building the project directly or building in VS.  We should either cross-compile with a TargetFramework values or include the XML in the `$(NetCoreAppCurrent)` configuration.